kubespawner/spawner.py

```diff
@@ -1424,6 +1424,7 @@ def get_state(self):
         """
         state = super().get_state()
         state['pod_name'] = self.pod_name
+        state['image'] = self.image
```

Comment:
This is sort of a hack, but what about adding
```
if hasattr(self, 'extra_state'):
    state['extra'] = self.extra_state
```
That way, if there's no explicitly defined attribute, nothing changes.  If someone defines `extra_state` in `kubespawner_override`, it'll be passed to state and can be used to cull.

But... this is a textbook definition of "quick hack that can't ever be removed"... any better ideas?

Reply:
Oh wow, sorry for the delay. Maybe? I think extra_state is supposed to be merged into the spawner by the time this function is called, but of course that means things not explicitly dumped will disappear. I'm not sure if there's a better solution. I'm not sure, but won't hasattr always return true because `extra_state` is a configurable? Might want to check hasattr and then if it's not None and/or empty?